tests/daemon/mock-persistent-cache.c

```diff
@@ -88,6 +88,7 @@ emer_persistent_cache_new (const gchar *directory,
   return self;
 }
 
+G_GNUC_NORETURN
 EmerPersistentCache *
 emer_persistent_cache_new_full (const gchar              *directory,
                                 guint64                   cache_size,
```

Comment:
Not sure if this is a good candidate for `G_GNUC_NORETURN`, `g_assert_not_reached` is supposed to be inert when compiled in release mode.

Maybe just `return NULL;` afterwards?

Reply:
Great point. I'll replace this with `g_return_val_if_reached (NULL);` which always returns the value, even in `G_DISABLE_CHECKS` mode.